staging/src/k8s.io/cli-runtime/pkg/genericclioptions/config_flags.go

```diff
@@ -284,7 +289,7 @@ func (f *ConfigFlags) toDiscoveryClient() (discovery.CachedDiscoveryInterface, e
 	httpCacheDir := filepath.Join(cacheDir, "http")
 	discoveryCacheDir := computeDiscoverCacheDir(filepath.Join(cacheDir, "discovery"), config.Host)
 
-	return diskcached.NewCachedDiscoveryClientForConfig(config, discoveryCacheDir, httpCacheDir, time.Duration(10*time.Minute))
+	return diskcached.NewCachedDiscoveryClientForConfig(config, discoveryCacheDir, httpCacheDir, cacheTTL)
```

Comment:
I agree with @jonnylangefeld here where he states that:
> The default can be increased though IMO.
I personally considered bumping that limit higher, as explained by @jonnylangefeld [here](https://github.com/kubernetes/kubernetes/pull/107141#discussion_r777165184) new resources which you'll try to fetch during that TTL will cause the cache invalidation and re-fetching all its contents. Further more, I've heard multiple complains from various customers about excessive discovery retrieval which are being then throttled both by the client and server. I'm not sure about whole day, but I think we can start with a few hours and re-visit that decision in a while. 

How about 6h for starters?

Reply:
done.